src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -651,7 +651,8 @@ BuildPlacementSelectList(uint32 groupId, List *relationShardList)
 		placement = FindShardPlacementOnGroup(groupId, relationShard->shardId);
 		if (placement == NULL)
 		{
-			ereport(ERROR, (errmsg("no active placement of shard %ld found on group %d",
```

Comment:
Wow, in some places `shardid` is signed, and in other places `shardid` is unsigned. We should fix this some day!

Reply:
I've been saying this for like three years.

Reply:
(Because we store it in a PostgreSQL table, and because PostgreSQL doesn't technically have unsigned types in SQL, I say it should be signed everywhere).

Reply:
opened #1849